### ISSUE TITLE
Repeated information on UI

### ISSUE THREAD

**Issue by paoloantinori** [custom-components/places#238]
Hi still see the bug described on #188

It kicks in with `Show last updated time at end of state '(since xx:yy)'` set to true.

And I'f I'm interpreting the code correctly the issue I think it's either here:

https://github.com/custom-components/places/blob/d40057cb3633c6b2effbeafb7b0469c38313a40f/custom_components/places/sensor.py#L2276

where the trimming happens only for a value that exceeds 256 chars

or

with the test conditions 

https://github.com/custom-components/places/blob/d40057cb3633c6b2effbeafb7b0469c38313a40f/custom_components/places/sensor.py#L2249

that should compare 2 locations for update only after stripping the ` (since xx:yy)` suffix

**Comment by Snuffy2** [comment]
I'm still confused about how this may be happening. Each update the status is rebuilt from scratch so I can't sort out how it may be adding multiple '(since xx:yy)' to the end of the state.

I don't believe it is either of the 2 code sections you listed above.

Line 2276 only trims the state string if it is larger than (225-14) to make sure when adding (since xx:yy) doesn't exceed the 255 character limit.

At the start of an update the previous state is saved prior to any updates. If show time is true, it is removed at line 2074.
https://github.com/custom-components/places/blob/d40057cb3633c6b2effbeafb7b0469c38313a40f/custom_components/places/sensor.py#L2074

Then the new state is built from scratch and compared to the previous value to decide if it needs to be updated or not.

Is this happening all of the time or just intermittently? Is it for all states or only when you are Home (or in a zone)?

**Comment by paoloantinori** [comment]
It doesn't happen constantly, but relatively frequently. I'm currently running a version of the code with additional logging calls to try to pinpoint what is causing this.
I'm gonna share update as soon as I have some

**Comment by Vinkoy** [comment]
I am observing the same issue when the person is in the zone or at home.
The new "since time" is appended to the existing string.
Home Assistant 2023.11.0 running in docker.

![image](https://github.com/custom-components/places/assets/18347539/e5fa72e4-4d01-4d02-b5d4-8e3549ef5f60)

**Comment by Snuffy2** [comment]
Since I've not been able to sort out what is causing this nor reproduce it on my machines, I'm not totally sure how to fix this. @paoloantinori did you ever find anything in your logging that might help?

All that being said, I did just release v2.5.3 that is more aggressive in removing the (since xx:yy) from the states and will remove >1 of them if they exist. Please give it a try and let me know if it fixes this.

If not, please [enable debug logging](https://github.com/custom-components/places/blob/master/CONTRIBUTING.md#enable-debug-logging-in-home-assistant) and post logs when this happens to see if I can sort out what is happening. If you want to email them instead of posting them (due to the location info in the logs), please send them to me at: my7idwq0@mail.snuffy2.com

**Comment by Snuffy2** [comment]
Hopefully this is now resolved. Please open a new issue if this is still happening.

**Comment by paoloantinori** [comment]
Yes, sorry for lack of updates. With your latest release the issue seems to
be gone! Thank you
